### ISSUE TITLE
Make kornia.enhance.solarize differentiable

### ISSUE THREAD

**Issue by paulgavrikov** [kornia/kornia#2490]
## 🚀 Feature
Make `kornia.enhance.solarize` differentiable.

## Motivation
Many if not all "enhance" operators are (naturally) differentiable, but `solarize` is not. 

## Pitch

1) It would be helpful to clarify that solarize is non-differentiable in the docs. My first impression about Kornia was that everything is differentiable.
2) There is next to no reasons, why this function is not differentiable. I assume the reason currently is the `torch.where` operator which could be replaced by a steep sigmoid tho, e.g. https://discuss.pytorch.org/t/differentiating-where-operator/59706/3 . Perhaps you could add a parameter to switch between the non-diff precise version and the sigmoid implementation.

## Additional context

Kornia: 0.6.12
Torch: 1.12.1

**Comment by edgarriba** [comment]
hi! do you want to differentiate respect to the input or the thresholds ? For the second, probably makes sense your suggestion. How to integrate in the same signature i'd say that if we can match same results I'm fine with it. /cc @shijianjian @johnnv1

**Comment by paulgavrikov** [comment]
@edgarriba yes, with respect to the thresholds

**Comment by shijianjian** [comment]
@paulgavrikov I think it makes sense to have this feature. Probably having a flag like `relaxation="none" | "sigmoid"`?

I think I did add some description for those non-differentiable transforms. Probably been removed somehow. I will add it later.

**Comment by shijianjian** [comment]
Hi @paulgavrikov, I am wondering if you are interested in throwing a PR?

**Comment by paulgavrikov** [comment]
@shijianjian currently out of office, but I’ll do that when I’m back (~2weeks)
